lmfit/parameter.py

```diff
@@ -611,10 +595,10 @@ def _getval(self):
                     self.value = self._expr_eval(self._expr_ast)
                     check_ast_errors(self._expr_eval)
 
-        try:
-            self.value = max(self.min, min(self._val, self.max))
-        except(TypeError, ValueError):
-            self.value = nan
+        v = self._val
+        if v > self.max: v = self.max
```

Comment:
It's the use of `min()` and `max()` functions that are the problem, not how
the test is done.

On Sun, Feb 21, 2016 at 9:22 PM, Andrew Nelson <notifications@github.com>
wrote:

> In lmfit/parameter.py
> <https://github.com/lmfit/lmfit-py/pull/303#discussion_r53583085>:
>
> > @@ -611,10 +595,10 @@ def _getval(self):
> >                      self.value = self._expr_eval(self._expr_ast)
> >                      check_ast_errors(self._expr_eval)
> >
> > -        try:
> > -            self.value = max(self.min, min(self._val, self.max))
> > -        except(TypeError, ValueError):
> > -            self.value = nan
> > +        v = self._val
> > +        if v > self.max: v = self.max
>
> I'd change these lines to:
>
> if self.min < self._val < self.max:
>     pass
> else:
>     self._val = max(self.min, min(self._val, self.max))
> self.value = self._val
>
> I like the range check form instead of two ifs. It's also faster.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lmfit/lmfit-py/pull/303/files#r53583085>.
>



-- 
--Matt Newville <newville at cars.uchicago.edu> 630-252-0431